lib/hocon/impl/simple_config.rb

```diff
@@ -107,6 +107,57 @@ def hash
     41 * @object.hash
   end
 
+  def self.find_key_or_null(me, key, expected, original_path)
+    v = me.peek_assuming_resolved(key, original_path)
+
+    if v.nil?
+      raise Hocon::ConfigError::ConfigMissingError.new(nil, original_path.render, nil)
+    end
+
+    if not expected.nil?
```

Comment:
I find `unless` to be really confusing, but I don't mind changing it if we want to stick with a certain pattern